x/ibc/handler_test.go

```diff
@@ -433,3 +433,199 @@ func (suite *HandlerTestSuite) TestHandleTimeoutPacket() {
+		name     string
+		malleate func()
+		expPass  bool
+	}{
+		{"success: ORDERED", func() {
+			clientA, clientB, connA, connB := suite.coordinator.SetupClientConnections(suite.chainA, suite.chainB, clientexported.Tendermint)
+			channelA, channelB := suite.coordinator.CreateChannel(suite.chainA, suite.chainB, connA, connB, channeltypes.ORDERED)
+			packet = channeltypes.NewPacket(suite.chainA.GetPacketData(suite.chainB), 1, channelA.PortID, channelA.ID, channelB.PortID, channelB.ID, timeoutHeight, 0)
+			counterpartyChannel = ibctesting.TestChannel{
+				PortID:               channelB.PortID,
+				ID:                   channelB.ID,
+				CounterpartyClientID: clientA,
+			}
+
+			// send from chainA to chainB
+			msg := ibctransfertypes.NewMsgTransfer(channelA.PortID, channelA.ID, ibctesting.TestCoin, suite.chainA.SenderAccount.GetAddress(), suite.chainB.SenderAccount.GetAddress().String(), packet.GetTimeoutHeight(), packet.GetTimeoutTimestamp())
+			err := suite.coordinator.SendMsgs(suite.chainA, suite.chainB, clientB, msg)
+			suite.Require().NoError(err) // message committed
+
+			// need to update chainA client to prove missing ack
+			suite.coordinator.UpdateClient(suite.chainA, suite.chainB, clientA, clientexported.Tendermint)
+
+			packetKey = host.KeyNextSequenceRecv(packet.GetDestPort(), packet.GetDestChannel())
+
+			// close counterparty channel
+			suite.coordinator.SetChannelClosed(suite.chainB, suite.chainA, counterpartyChannel)
+
+		}, true},
+		{"success: UNORDERED", func() {
+			clientA, clientB, _, _, channelA, channelB := suite.coordinator.Setup(suite.chainA, suite.chainB)
+			packet = channeltypes.NewPacket(suite.chainA.GetPacketData(suite.chainB), 1, channelA.PortID, channelA.ID, channelB.PortID, channelB.ID, timeoutHeight, 0)
+			counterpartyChannel = ibctesting.TestChannel{
+				PortID:               channelB.PortID,
+				ID:                   channelB.ID,
+				CounterpartyClientID: clientA,
+			}
+
+			// send from chainA to chainB
+			msg := ibctransfertypes.NewMsgTransfer(channelA.PortID, channelA.ID, ibctesting.TestCoin, suite.chainA.SenderAccount.GetAddress(), suite.chainB.SenderAccount.GetAddress().String(), packet.GetTimeoutHeight(), packet.GetTimeoutTimestamp())
+			err := suite.coordinator.SendMsgs(suite.chainA, suite.chainB, clientB, msg)
+			suite.Require().NoError(err) // message committed
+
+			// need to update chainA client to prove missing ack
+			suite.coordinator.UpdateClient(suite.chainA, suite.chainB, clientA, clientexported.Tendermint)
+
+			packetKey = host.KeyPacketAcknowledgement(packet.GetDestPort(), packet.GetDestChannel(), packet.GetSequence())
+
+			// close counterparty channel
+			suite.coordinator.SetChannelClosed(suite.chainB, suite.chainA, counterpartyChannel)
+
+		}, true},
+		{"success: UNORDERED timeout out of order packet", func() {
+			// setup uses an UNORDERED channel
+			clientA, clientB, _, _, channelA, channelB := suite.coordinator.Setup(suite.chainA, suite.chainB)
```

Comment:
nit, without the comment, it is not intuitive that the channels are unordered. Maybe consider renaming it or adding a boolean arg?

Reply:
I agree, I'm thinking we will likely have more channel types in the future, so a bool or different function names wouldn't make sense. Instead I could add a third arg that would be like `channeltypes.ORDERED`. This is a fairly large line diff change and I think would be easier to review in a followup, I can add it to the wishlist and address it when I begin to untangle `ibc-transfer` from testing.